pkg/agent/client.go

```diff
@@ -248,36 +248,36 @@ func (a *AgentClient) initializeAuthContext(ctx context.Context) (context.Contex
 // The requests include things like opening a connection to a server,
 // streaming data and close the connection.
 func (a *AgentClient) Serve() {
-	klog.Infof("Start serving for serverID %s", a.serverID)
+	klog.V(2).InfoS("Start serving", "serverID", a.serverID)
 	go a.probe()
 	for {
 		select {
 		case <-a.stopCh:
-			klog.Info("stop agent client.")
+			klog.V(2).Infoln("stop agent client.")
 			return
 		default:
 		}
 
 		pkt, err := a.Recv()
 		if err != nil {
 			if err == io.EOF {
-				klog.Info("received EOF, exit")
+				klog.V(2).Infoln("received EOF, exit")
 				return
 			}
-			klog.Warningf("stream read error: %v", err)
+			klog.ErrorS(err, "could not read stream")
 			return
 		}
 
-		klog.Infof("[tracing] recv packet, type: %s", pkt.Type)
+		klog.V(5).InfoS("[tracing] recv packet", "type", pkt.Type)
 
 		if pkt == nil {
-			klog.Warningf("empty packet received")
+			klog.V(2).Infoln("empty packet received")
```

Comment:
Wondering if this should be V(3) or V(4). Not clear this corresponds to a "significant state change in the system". I don't think we're expecting empty packets but I could see us drowning under the log level, given a bug. If we make it at least V(3), then we can turn it off if we find an issue.